cmd/post.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright © 2022 himkt <himkt@klis.tsukuba.ac.jp>
+
+*/
+package cmd
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"path"
+
+	"github.com/spf13/cobra"
+)
+
+type Payload struct {
+	Content string `json:"content"`
+}
+
+// postCmd represents the post command
+var postCmd = &cobra.Command{
+	Use:   "post",
+	Short: "Post text to discord server",
+	Long:  `Post text to discord server using HTTP.`,
+	Run: func(cmd *cobra.Command, args []string) {
+		home, _ := os.UserHomeDir()
+		configPath := path.Join(home, ".config/dobato/webhook")
+		if _, err := os.Stat(configPath); errors.Is(err, os.ErrNotExist) {
+			fmt.Println("No config available, run dobato setup first.")
+			return
+		}
+
+		f, err := os.Open(configPath)
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+		}
+
+		defer f.Close()
```

Comment:
I'm not sure what `defer` does. 🙄

Reply:
defer is execute function after function return or panic.

https://go.dev/play/p/2gDIq9M9hbJ

Reply:
Thank you, it basically makes sense!

Q. Is `after function return or panic` right? It looks defered lines are executed just before a function return or panic.

https://go.dev/play/p/4kk1lvjG4eM

Reply:
> Q. Is after function return or panic right?
No, I made a mistake 🙇 

> It looks defered lines are executed just before a function return or panic.
It's correct.